pkg/store/consul/kv.go

```diff
@@ -232,6 +232,45 @@ func (c consulStore) SetPodTxn(ctx context.Context, podPrefix PodPrefix, nodenam
 	})
 }
 
+// LockIfKeyNotExistsTxn will lock and set a key, guaranteeing the key did not
+// exist.
+func (c consulStore) LockIfKeyNotExistsTxn(
+	ctx context.Context,
+	podPrefix PodPrefix,
+	nodename types.NodeName,
+	manifest manifest.Manifest,
+	session Session,
+) error {
+	buf := bytes.Buffer{}
+	err := manifest.Write(&buf)
+	if err != nil {
+		return err
+	}
+
+	key, err := podPath(podPrefix, nodename, manifest.ID())
+	if err != nil {
+		return err
+	}
+
+	// KVDeleteCAS with index 0 is used because it ensures that the subsequent
+	// lock will fail if the key existed.
+	err = transaction.Add(ctx, api.KVTxnOp{
```

Comment:
might be useful to add a `TODO` for changing this to `check-not-exists` once we upgrade consul

Reply:
don't worry about this, don't want to slow you down